hardware_interface/src/resource_manager.cpp

```diff
@@ -1194,7 +1257,13 @@ void ResourceManager::import_controller_exported_state_interfaces(
   const std::string & controller_name, std::vector<StateInterface> & interfaces)
 {
   std::lock_guard<std::recursive_mutex> guard(resource_interfaces_lock_);
-  auto interface_names = resource_storage_->add_state_interfaces(interfaces);
+  std::vector<StateInterface::SharedPtr> interface_ptrs;
+  interface_ptrs.reserve(interfaces.size());
+  for (auto & interface : interfaces)
+  {
+    interface_ptrs.push_back(std::make_shared<StateInterface>(interface));
+  }
+  auto interface_names = resource_storage_->add_state_interfaces(interface_ptrs);
```

Comment:
can we use this method also for hardware? If yes, then we can avoid completely having `add_state_interface` and compatible method for commands?